src/Http/Middleware/DoublePassDecoratorMiddleware.php

```diff
@@ -55,6 +58,10 @@ class DoublePassDecoratorMiddleware implements MiddlewareInterface
      */
     public function __construct(callable $callable)
     {
+        deprecationWarning(
+            '"Double pass" middleware are deprecated.'
+            . ' Use a `Closure` or a class which implements `Psr\Http\Server\MiddlewareInterface` instead.'
```

Comment:
```suggestion
            . ' Use a `Closure` with the signature of `($request, $handler)` or a class which implements `Psr\Http\Server\MiddlewareInterface` instead.'
```

Should we clarify the expected arity of the closure as 3 parameter middlewares are deprecated.

Reply:
I think stating the required signature for Closures should suffice. The error message would get too long otherwise. The deprecated signature is mentioned in the migration guide.